improver_tests/psychrometric_calculations/test_PhaseChangeLevel.py

```diff
@@ -418,6 +418,140 @@ def test_null_lat_lon(self):
         self.assertArrayAlmostEqual(result.data, expected_data)
 
 
+class Test_horizontally_interpolate_phase(IrisTest):
+
+    """Test the PhaseChangeLevel horizontal interpolation."""
+
+    def setUp(self):
+        """Set up input data."""
+
+        # A simple 1d case.
+        self.phase_change_data_1d = np.array([[1000.0, np.nan, 800.0]])
+        self.orography_1d = np.array([[850.0, 700.0, 500.0]])
+        self.max_nbhood_orog_1d = np.array([[850.0, 850.0, 700.0]])
+        self.expected_result_1d = np.array([[1000.0, 1000.0, 800.0]])
+
+        # A case that mimics a real side-of-mountain failure.
+        self.phase_change_data_2d = np.array(
+            [
+                [1000.0, 1000.0, 950.0, 800.0, 800.0],
+                [1000.0, 1000.0, 950.0, 900.0, 800.0],
+                [1000.0, 1000.0, 950.0, np.nan, 800.0],
+                [1000.0, 1000.0, 950.0, 900.0, 800.0],
+                [1000.0, 1000.0, 950.0, 800.0, 800.0],
+            ]
+        )
+        self.orography_2d = np.array(
+            [
+                [400.0, 500.0, 500.0, 500.0, 400.0],
+                [500.0, 700.0, 750.0, 700.0, 500.0],
+                [500.0, 700.0, 850.0, 700.0, 500.0],
+                [500.0, 700.0, 750.0, 700.0, 500.0],
+                [400.0, 500.0, 500.0, 500.0, 400.0],
+            ]
+        )
+        self.max_nbhood_orog_2d = np.array(
+            [
+                [700.0, 700.0, 700.0, 700.0, 700.0],
+                [700.0, 850.0, 850.0, 850.0, 700.0],
+                [700.0, 850.0, 850.0, 850.0, 700.0],
+                [700.0, 850.0, 850.0, 850.0, 700.0],
+                [700.0, 700.0, 700.0, 700.0, 700.0],
+            ]
+        )
+        self.expected_result_2d = np.array(
+            [
+                [1000.0, 1000.0, 950.0, 800.0, 800.0],
+                [1000.0, 1000.0, 950.0, 900.0, 800.0],
+                [1000.0, 1000.0, 950.0, 950.0, 800.0],
+                [1000.0, 1000.0, 950.0, 900.0, 800.0],
+                [1000.0, 1000.0, 950.0, 800.0, 800.0],
+            ]
+        )
+
+        # A 'nan crater' with low orography circled by high orography.
+        self.phase_change_data_2d_crater = np.full((9, 9), 1000.0)
+        self.phase_change_data_2d_crater[2:7, 2:7] = np.nan
+        self.orography_2d_crater = np.full((9, 9), 900.0)
+        self.orography_2d_crater[2:7, 2:7] = 600.0
+        self.max_nbhood_orog_2d_crater = np.full((9, 9), 900.0)
+        self.max_nbhood_orog_2d_crater[3:6, 3:6] = 600.0
+        self.expected_result_2d_crater = np.full((9, 9), 1000.0)
+
+    def test_interpolate_edge_case_1d(self):
+        """Test that we still fill in missing areas under a 1d peaked edge case."""
```

Comment:
The word "still", in all these test doc-strings, is only appropriate for this PR. It probably ought to be removed for future clarity.